search/api.py

```diff
@@ -69,29 +76,33 @@ def create_search_obj(user, search_param_dict=None, filter_on_email_optin=False)
     Args:
         user (User): User object
         search_param_dict (dict): A dict representing the body of an ES query
+        filter_on_email_optin (bool): If true, filter out profiles where email_optin != True
 
     Returns:
         Search: elasticsearch_dsl Search object
     """
     search_obj = Search(index=settings.ELASTICSEARCH_INDEX, doc_type=DOC_TYPES)
-    # the following filter should come first because the sequence matters in applying them
+    # Update from search params first so our server-side filtering will overwrite it if necessary
+    if search_param_dict is not None:
+        search_obj.update_from_dict(search_param_dict)
+        # Early versions of searchkit use filter which isn't in the DSL but it acts equivalently to post_filter
+        if 'filter' in search_param_dict:
+            search_obj = search_obj.post_filter(search_param_dict['filter'])
```

Comment:
I figured it was cleaner to do it using the search object than to edit the dictionary. I also wanted to leave `search_param_dict` unmodified since it's being passed as an input value